components/form/TaskForm.tsx

```diff
@@ -36,6 +36,110 @@ const validateTaskFields = (fields: TaskFields, projectId: number): Task => {
     throw Error("Invalid task form: missing required fields")
 }
 
+type TaskFieldInputProps = {
+    taskFields: TaskFields
+    setTaskFields: React.Dispatch<React.SetStateAction<TaskFields>>
+}
+
+export function NameInput({
+    taskFields,
+    setTaskFields,
+}: TaskFieldInputProps): JSX.Element {
+    return (
+        <FormControl isInvalid={!taskFields.name} isRequired>
+            <FormLabel>Task Name</FormLabel>
+            <Input
+                placeholder="Task Name"
+                onChange={(event) =>
+                    setTaskFields({
+                        ...taskFields,
+                        name: event.target.value,
+                    })
+                }
+            />
+            <FormErrorMessage>Task name cannot be empty.</FormErrorMessage>
```

Comment:
Should this be rendered conditionally?

Reply:
I think that's a chakra component that's only rendered if the isInvalid attribute equals true in the parent component